File: src/cli/commands/doctor.js
import fs from 'node:fs';
import path from 'node:path';
import chalk from 'chalk';
import { findProjectRoot, loadConfig } from '../../lib/config.js';

export function registerDoctorCommand(program) {
  program
    .command('doctor')
    .description('Check project health')
    .action(() => {
      const root = findProjectRoot();
      if (!root) {
        console.error(chalk.red('No .holoctl/ found. Run `holoctl init` first.'));
        process.exit(1);
      }

      console.log(chalk.bold('\n  holoctl doctor\n'));
      let issues = 0;

      // Check config
      try {
        const config = loadConfig(root);
        check('Config', '.holoctl/config.json is valid', true);
      } catch (e) {
        check('Config', `.holoctl/config.json: ${e.message}`, false);
        issues++;
      }

      // Check board
      const indexPath = path.join(root, '.holoctl', 'board', 'index.json');
      if (fs.existsSync(indexPath)) {
        try {
          const data = JSON.parse(fs.readFileSync(indexPath, 'utf8'));
          const ticketCount = data.tickets?.length || 0;
          check('Board', `index.json valid (${ticketCount} tickets)`, true);
        } catch (e) {
          check('Board', `index.json parse error: ${e.message}`, false);
          issues++;
        }
      } else {
        check('Board', 'index.json exists', false);
        issues++;
      }

      // Check agents
      const agentsDir = path.join(root, '.holoctl', 'agents');
      if (fs.existsSync(agentsDir)) {
        const agents = fs.readdirSync(agentsDir).filter(f => f.endsWith('.md'));
        check('Agents', `${agents.length} agent(s) defined`, agents.length > 0);
        if (agents.length === 0) issues++;
      } else {
        check('Agents', 'agents/ directory exists', false);
        issues++;
      }

      // Check commands
      const commandsDir = path.join(root, '.holoctl', 'commands');
      if (fs.existsSync(commandsDir)) {
        const commands = fs.readdirSync(commandsDir).filter(f => f.endsWith('.md'));
        check('Commands', `${commands.length} command(s) defined`, commands.length > 0);
        if (commands.length === 0) issues++;
      } else {
        check('Commands', 'commands/ directory exists', false);
        issues++;
      }

      // Check instructions
      const instructionsPath = path.join(root, '.holoctl', 'instructions.md');
      check('Instructions', 'instructions.md exists', fs.existsSync(instructionsPath));
      if (!fs.existsSync(instructionsPath)) issues++;

      // Check context
      const contextDir = path.join(root, '.holoctl', 'context');
      check('Context', 'context/ directory exists', fs.existsSync(contextDir));
      if (!fs.existsSync(contextDir)) issues++;

      console.log('');
      if (issues === 0) {
        console.log(chalk.green('  All checks passed. Project is healthy.\n'));
      } else {
        console.log(chalk.yellow(`  ${issues} issue(s) found.\n`));
      }
    });
}

function check(category, message, ok) {
  const icon = ok ? chalk.green('✓') : chalk.red('✗');
  console.log(`  ${icon} ${chalk.dim(category.padEnd(14))} ${message}`);
}

File: projctl/cli/doctor.py
from __future__ import annotations
import json

import typer
from ._console import console

from ..lib.config import find_project_root, load_config

app = typer.Typer()


@app.command("doctor")
def doctor_cmd():
    """Check project health."""
    root = find_project_root()
    if not root:
        console.print("[red]No .projctl/ found. Run `projctl init` first.[/red]")
        raise typer.Exit(1)

    console.print("\n  [bold]projctl doctor[/bold]\n")
    issues = 0

    try:
        load_config(root)
        _check("Config", ".projctl/config.json is valid", True)
    except Exception as e:
        _check("Config", f".projctl/config.json: {e}", False)
        issues += 1

    index_path = root / ".projctl" / "board" / "index.json"
    if index_path.exists():
        try:
            data = json.loads(index_path.read_text(encoding="utf-8"))
            count = len(data.get("tickets", []))
            _check("Board", f"index.json valid ({count} tickets)", True)
        except Exception as e:
            _check("Board", f"index.json parse error: {e}", False)
            issues += 1
    else:
        _check("Board", "index.json exists", False)
        issues += 1

    agents_dir = root / ".projctl" / "agents"
    if agents_dir.exists():
        agent_count = len(list(agents_dir.glob("*.md")))
        ok = agent_count > 0
        _check("Agents", f"{agent_count} agent(s) defined", ok)
        if not ok:
            issues += 1
    else:
        _check("Agents", "agents/ directory exists", False)
        issues += 1

    commands_dir = root / ".projctl" / "commands"
    if commands_dir.exists():
        cmd_count = len(list(commands_dir.glob("*.md")))
        ok = cmd_count > 0
        _check("Commands", f"{cmd_count} command(s) defined", ok)
        if not ok:
            issues += 1
    else:
        _check("Commands", "commands/ directory exists", False)
        issues += 1

    instructions_path = root / ".projctl" / "instructions.md"
    ok = instructions_path.exists()
    _check("Instructions", "instructions.md exists", ok)
    if not ok:
        issues += 1

    context_dir = root / ".projctl" / "context"
    ok = context_dir.exists()
    _check("Context", "context/ directory exists", ok)
    if not ok:
        issues += 1

    console.print("")
    if issues == 0:
        console.print("[green]  All checks passed. Project is healthy.[/green]\n")
    else:
        console.print(f"[yellow]  {issues} issue(s) found.[/yellow]\n")


def _check(category: str, message: str, ok: bool) -> None:
    icon = "[green]✓[/green]" if ok else "[red]✗[/red]"
    console.print(f"  {icon} [dim]{category:<14}[/dim] {message}")
